akka-http-core/src/main/scala/akka/http/impl/engine/http2/client/PersistentConnection.scala

```diff
@@ -128,12 +139,32 @@ private[http2] object PersistentConnection {
             failStage(new RuntimeException(s"Connection failed after $maxAttempts attempts", cause))
           } else {
             setHandler(requestIn, Unconnected)
-            log.info(s"Connection attempt failed: ${cause.getMessage}. Trying to connect again${connectsLeft.map(n => s" ($n attempts left)").getOrElse("")}.")
-            connect(connectsLeft)
+            val nextEmbargo = embargo match {
+              case Duration.Zero => baseEmbargo
+              case otherValue    => (otherValue * 2).min(maxBaseEmbargo)
+            }
+            if (nextEmbargo == Duration.Zero) {
+              log.info(s"Connection attempt failed: ${cause.getMessage}. Trying to connect again${connectsLeft.map(n => s" ($n attempts left)").getOrElse("")}.")
+              connect(connectsLeft, nextEmbargo)
+            } else {
+              // FIXME is it fine to just stay in Connecting state here while backing off?
```

Comment:
I guess, it's fine, the states basically describe the state in relation to outside communication, so we could see the backoff as a delayed "connecting" phase.

Reply:
I was wondering if there is any up/downstream stop/failure handling that would need to be specific to the backoff, but maybe not.